File: lib/core/rfid/rfid_tag.dart
class RfidTag {
  const RfidTag({
    required this.epc,
    required this.timestamp,
    this.tid,
    this.rssi,
    this.count = 1,
  });

  final String epc;
  final String? tid;
  final int? rssi;
  final DateTime timestamp;
  final int count;

  RfidTag copyWith({
    String? epc,
    String? tid,
    int? rssi,
    DateTime? timestamp,
    int? count,
  }) {
    return RfidTag(
      epc: epc ?? this.epc,
      tid: tid ?? this.tid,
      rssi: rssi ?? this.rssi,
      timestamp: timestamp ?? this.timestamp,
      count: count ?? this.count,
    );
  }
}

File: rfid-app/app/src/main/java/com/smartx/rfidreader/core/reader/RfidTag.kt
package com.smartx.rfidreader.core.reader

import java.util.Date

data class RfidTag(
    val epc: String,
    val rssi: String = "",
    val tid: String = "",
    val user: String = "",
    val antenna: Int = 1,
    val readCount: Int = 1,
    val timestamp: Date = Date()
)
